crates/modelardb_server/src/storage/uncompressed_data_manager.rs

```diff
@@ -354,44 +356,35 @@ impl UncompressedDataManager {
 
     // TODO: Remove this when compression component is changed.
     /// Compress the full [`UncompressedInMemoryDataBuffer`] given as `in_memory_data_buffer` and
-    /// return the resulting compressed and merged segments as a [`RecordBatch`].
+    /// return the resulting compressed and merged segments as a [`RecordBatch`]. Returns
+    /// [`IOError`] if the error bound cannot be retrieved from the [`MetadataManager`].
     async fn compress_finished_buffer(
         &mut self,
         univariate_id: u64,
         mut in_memory_data_buffer: UncompressedInMemoryDataBuffer,
-    ) -> RecordBatch {
+    ) -> Result<RecordBatch, IOError> {
         // Add the size of the segment back to the remaining reserved bytes and record the change.
         let freed_memory = UncompressedInMemoryDataBuffer::memory_size();
         self.uncompressed_remaining_memory_in_bytes += freed_memory;
         self.used_uncompressed_memory_metric
             .append(-(freed_memory as isize), true);
 
-        // unwrap() is safe to use since the error bound is not negative, infinite, or NAN.
-        let error_bound = ErrorBound::try_new(0.0).unwrap();
-
         // unwrap() is safe to use since get_record_batch() can only fail for spilled buffers.
```

Comment:
This is not changed in this PR but this should be changed to `record_batch()`.

Reply:
Fixed.